frontend/public/components/terminal.jsx

```diff
@@ -116,8 +115,11 @@ class Terminal_ extends React.Component {
       this.fitAddon.fit();
       // update the pty
       this.props.onResize(terminal.rows, terminal.cols);
+      // The internal xterm textarea was not repositioned when the window was resized.
+      // This workaround triggers an position update.
+      this.terminal._core?._syncTextArea?.();
```

Comment:
Correct me if I'm wrong, but this seems to be the only line needed to fix the bug. I would prefer that the rest of the changes (besides your comments here) be removed from the PR as they don't seem to be relevant to this bug fix.

Reply:
@TheRealJon your right, and it also make sense to keep this PR small. I dropped all other changes and retested the fix, still works fine. PTAL.